### ISSUE TITLE
Documentation lacks information for the format of the json config files used. 

### ISSUE THREAD

**Issue by lchausmann** [codemac/gcalorg#1]
Nice project, it might be just the thing I am looking for. Could you give me a hint on how the config should look like?

**Comment by codemac** [comment]
The first time you run it, it should generate those json files for you, the paths are obviously crap because I haven't really written this for other people to use - though you raise a great point. I'll try to see if I can't clean some stuff up about it.

I'm also working on it handling recursive tasks correctly. Currently it depends on the v3 api for exapnding those events out, instead of using org-mode syntax.

**Comment by lchausmann** [comment]
Thank you! 

I got inspired by you code and ended up hacking something very similar together, but based it on cobra and added some configs for it. It's here: https://github.com/lchausmann/gcalsync 

I've added a mentions to you, if that's not alright let me know and I will take it all off line and remove it.

**Comment by codemac** [comment]
If you're willing to merge the projects, it's something I'm more than happy to do if we agree on licensing. My code ideally would be gpl, but I'm willing to bend on that if you have strong feelings.

I'd like to keep the org printing & event handling centralized so we can both profit from eachother's improvements in either case. I just added better repeated event handling.

**Comment by lchausmann** [comment]
I dont have a strong thing about license. So as long as it's open, I dont mind one way or the other. I just stuck with the MIT license suggested by the cobra CLI tool.

Regarding the merging, that'd be super cool. It was just something bigger, so I didnt want to force it on you.

I'd find it to be an excellent idea, if we merged it. I've moved a bit of the code around, but I for sure wouldnt mind merging it into your repo, or providing you access to to the one, I am currently working on. 

The org file  handling is mostly yours, I've just moved it's own file, change some points to match with the struct I ended up parsing configs into, the rest is all yours.

But if that's ok with you, I don't think it would be to much work to merge the two, and I would be happy to start on it soonish. This week is a bit full, but some time soon no doubt.

So should I move things around, and submit a pull request?

**Comment by codemac** [comment]
* lchausmann <notifications@github.com> [2018-03-05 07:32]:
> I dont have a strong thing about license. So as long as it's 
> open, I dont mind one way or the other.

Ok - I will be putting this all under GPL3, you can either license 
your contributions under GPL3 or refactor them into a separate 
file for a GPL compatible license if required.
 
> Regarding the merging, that'd be super cool. It was just 
> something bigger, so I didnt want to force it on you. 
> 
> I'd find it to be an excellent idea, if we merged it. I've moved 
> a bit of the code around, but I for sure wouldnt mind merging it 
> into your repo, or providing you access to to the one, I am 
> currently working on.  
> 
> The org file  handling is mostly yours, I've just moved it's own 
> file, change some points to match with the struct I ended up 
> parsing configs into.  
> 
> But if that's ok with you, I don't think it would be to much 
> work to merge the two, and I would be happy to start on it 
> soonish. This week is a bit full, but some time soon no doubt. 
> 
> So should I move things around, and submit a pull request? 

I pushed my most recent moves around - the repo as it stands is go 
gettable and a pull request would be completely reasonable.

Feel free to make small tiny improvements, and throw pull requests my way. I will do my best to respond quickly, and by default I merge :)

**Comment by lchausmann** [comment]
Jeff Mickey <notifications@github.com> writes:

> * lchausmann <notifications@github.com> [2018-03-05 07:32]:
>> I dont have a strong thing about license. So as long as it's
>> open, I dont mind one way or the other.
>
> Ok - I will be putting this all under GPL3, you can either 
> license
> your contributions under GPL3 or refactor them into a separate
> file for a GPL compatible license if required.

Thats fine.

[...]

> I pushed my most recent moves around - the repo as it stands is 
> go
> gettable and a pull request would be completely reasonable.
>
> Feel free to make small tiny improvements, and throw pull 
> requests my way. I will do my best to respond quickly, and by 
> default I merge :)

I would like to move it into a cobra based cli, since it provides 
a ton
of help in the config department, so I would likely move some 
stuff into
a cmd pacakge, if that's ok with you, I will start on a pull 
request
then?

/LCH

**Comment by codemac** [comment]
Yes! I was just looking through what you did today in your repo, yes please :)

**Comment by lchausmann** [comment]
Hi,

That's sweet, I will do the same on your repo and submit a pull 
request,
when I am done.


Jeff Mickey <notifications@github.com> writes:

> Yes! I was just looking through what you did today in your repo, 
> yes please :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the 
> thread.


--
Dbh,
Lars Chr.
